selfupdate/detect.go

```diff
@@ -27,18 +27,26 @@ func findSuitableReleaseAndAsset(rels []*github.RepositoryRelease) (*github.Repo
 	}
 
 	for _, rel := range rels {
-		if rel.GetDraft() {
+		if targetVersion == "" && rel.GetDraft() {
 			log.Println("Skip draft version", rel.GetTagName())
 			continue
 		}
-		if rel.GetPrerelease() {
+
+		if targetVersion == "" && rel.GetPrerelease() {
 			log.Println("Skip pre-release version", rel.GetTagName())
 			continue
 		}
+
 		if !reVersion.MatchString(rel.GetTagName()) {
 			log.Println("Skip version not adopting semver", rel.GetTagName())
 			continue
 		}
+
+		if targetVersion != "" && targetVersion != rel.GetTagName() {
+			log.Println("Skip not matching version", rel.GetTagName())
+			continue
+		}
```

Comment:
Please move this `if` statement to the top of this loop. Then I think `targetVersion == ""` checks will not be needed.